src/app/charts/line-graph.component.ts

```diff
@@ -131,22 +134,19 @@ export class LineGraphComponent {
     private drawTrendLine(): void {
         // Only draw if data and add trendline flag
         if (this.trendline && this.extractedData.length) {
-          // Set up if first time
-          if (!this.trendData) {
```

Comment:
I'd put a check for trendData before to prevent unnecessary recalculations. Since data is no longer static, we'll want to recalculate lines.